Source/ACE.Server/WorldObjects/Player_Magic.cs

```diff
@@ -1001,8 +1001,9 @@ public bool CreatePlayerSpell(WorldObject target, TargetCategory targetCategory,
             var caster = casterItem ?? GetEquippedWand();
             var isWeaponSpell = casterItem != null && IsWeaponSpell(spell.Id, casterItem);
 
-            // Grab player's skill level in the spell's Magic School
-            var magicSkill = GetCreatureSkill(spell.School).Current;
+            // Grab player's skill level in the spell's Magic School (matches CACQualities::InqSkillLevel)
+            var playerSkill = GetCreatureSkill(spell.School);
+            var magicSkill = playerSkill.InitLevel + playerSkill.Ranks;
```

Comment:
You can't change magicSkill here. This effectively means a player's buffs will be useless for all magic calculations from this point forward.

Reply:
gotcha